seed/csharp-model/grpc-proto/src/SeedApi/Core/ProtoConverter.cs

```diff
@@ -0,0 +1,64 @@
+using Google.Protobuf.WellKnownTypes;
+
+#nullable enable
+
+namespace SeedApi;
+
+/// <summary>
+/// Utility class for converting to and from Protobuf types.
+/// </summary>
+internal class ProtoConverter
+{
+    public static Struct ToProtoStruct(Dictionary<string, MetadataValue?> value)
+    {
+        var result = new Struct();
+        foreach (var kvp in value)
+        {
+            result.Fields[kvp.Key] = ToProtoValue(kvp.Value);
+        }
+        return result;
+    }
+
+    public static Dictionary<string, MetadataValue?> FromProtoStruct(Struct value)
+    {
+        var result = new Dictionary<string, MetadataValue?>();
+        foreach (var kvp in value.Fields)
+        {
+            result[kvp.Key] = FromProtoValue(kvp.Value);
+        }
+        return result;
+    }
+
+    public static Value ToProtoValue(MetadataValue? value)
+    {
+        if (value == null)
+        {
+            Value.ForNull();
```

Comment:
should it be:
```suggestion
            return Value.ForNull();
```

Reply:
Definitely, great catch - thanks for the keen eye